registration-processor/registration-processor-info-storage-service/src/main/java/io/mosip/registration/processor/packet/storage/utils/ABISHandlerUtil.java

```diff
@@ -1,222 +1,228 @@
+import io.mosip.registration.processor.status.code.RegistrationStatusCode;
+import io.mosip.registration.processor.status.dto.InternalRegistrationStatusDto;
+import io.mosip.registration.processor.status.dto.SyncTypeDto;
+
+/**
+ * The Class ABISHandlerUtil.
+ * 
+ * @author Nagalakshmi
+ * @author Horteppa
+ */
+@Component
+public class ABISHandlerUtil {
+	
+	/** The reg proc logger. */
+	private static Logger regProcLogger = RegProcessorLogger.getLogger(ABISHandlerUtil.class);
+
+	/** The utilities. */
+	@Autowired
+	Utilities utilities;
+
+	/** The packet info manager. */
+	@Autowired
+	private PacketInfoManager<Identity, ApplicantInfoDto> packetInfoManager;
+
+	/** The packet info dao. */
+	@Autowired
+	private PacketInfoDao packetInfoDao;
+
+	@Autowired
+	private IdRepoService idRepoService;
+
+	/**
+	 * Gets the unique reg ids.
+	 *
+	 * @param registrationId   the registration id
+	 * @param registrationType the registration type
+	 * @return the unique reg ids
+	 * @throws ApisResourceAccessException           the apis resource access
+	 *                                               exception
+	 * @throws IOException                           Signals that an I/O exception
+	 *                                               has occurred.
+	 * @throws                                       io.mosip.kernel.core.exception.IOException
+	 */
+	public List<String> getUniqueRegIds(String registrationId, String registrationType, ProviderStageName stageName) throws ApisResourceAccessException, JsonProcessingException, PacketManagerException, IOException {
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
+				registrationId, "ABISHandlerUtil::getUniqueRegIds()::entry");
+		
+		String latestTransactionId = utilities.getLatestTransactionId(registrationId);
+
+		List<String> regBioRefIds = packetInfoDao.getAbisRefMatchedRefIdByRid(registrationId);
+
+		List<String> machedRefIds = new ArrayList<>();
+		List<String> uniqueRIDs = new ArrayList<>();
+		List<AbisResponseDetDto> abisResponseDetDtoList = new ArrayList<>();
+
+		if (!regBioRefIds.isEmpty()) {
+			List<AbisResponseDto> abisResponseDtoList = packetInfoManager.getAbisResponseRecords(regBioRefIds.get(0),
+					latestTransactionId, AbisConstant.IDENTIFY);
+			for (AbisResponseDto abisResponseDto : abisResponseDtoList) {
+				abisResponseDetDtoList.addAll(packetInfoManager.getAbisResponseDetails(abisResponseDto.getId()));
+			}
+			if (!abisResponseDetDtoList.isEmpty()) {
+				for (AbisResponseDetDto abisResponseDetDto : abisResponseDetDtoList) {
+					machedRefIds.add(abisResponseDetDto.getMatchedBioRefId());
+				}
+				if (!CollectionUtils.isEmpty(machedRefIds)) {
+					List<String> matchedRegIds = packetInfoDao.getAbisRefRegIdsByMatchedRefIds(machedRefIds);
+					if (!CollectionUtils.isEmpty(matchedRegIds)) {
+						List<String> processingRegIds = packetInfoDao.getWithoutStatusCodes(matchedRegIds,

```

Comment:
If the current status of the packet is FAILED or REPROCESS_FAILED, will it not get to this list ?

Reply:
Yeah it will come to this list also.I added fix on friday and again reverted as you suggested to do in processed records.

Reply:
Code fixed